apps/dh/api-dh/source/DataHub.WebApi/GraphQL/Mutation.cs

```diff
@@ -123,17 +123,48 @@ await client
         [Service] IEdiB2CWebAppClient_V1 client)
     {
         await client.RequestAggregatedMeasureDataAsync(
-            new RequestAggregatedMeasureDataMarketRequest()
-            {
-                ProcessType = processType,
-                MeteringPointType = meteringPointType,
-                StartDate = startDate,
-                EndDate = endDate,
-                GridArea = gridArea,
-                EnergySupplierId = energySupplierId,
-                BalanceResponsibleId = balanceResponsibleId,
-            },
-            cancellationToken)
+                "1.0",
+                new RequestAggregatedMeasureDataMarketRequest()
+                {
+                    CalculationType = calculationType,
+                    MeteringPointType = meteringPointType!.Value,
+                    StartDate = startDate,
+                    EndDate = endDate,
+                    GridArea = gridArea,
+                    EnergySupplierId = energySupplierId,
+                    BalanceResponsibleId = balanceResponsibleId,
+                },
+                cancellationToken)
+            .ConfigureAwait(false);
+        return true;
+    }
+
+    public async Task<bool> CreateWholesaleSettlementRequestAsync(
+        EdiB2CWebAppCalculationType calculationType,
+        string startDate,
+        string? endDate,
```

Comment:
suggestion: use NodaTime.Interval here instead of separate arguments for start and end. we have logic that automatically "translates" frontend intervals to backend intervals (they are not fully equal).

Reply:
Should we make this particular interval change now, or as part of the changes mentioned above, and thus in a later PR?

Reply:
Preferably in this PR, since you are the expert on how to serialize the values in accordance with what the backend client expects 🙂